### PR TITLE
Imp 007 polish classes

### PR DESCRIPTION
**Pull request by jackyzha0**

### CODE REVIEW

**Comment by jackyzha0** [comment]
Should fix #11

File: IMP/creatures.py [hunk 1]
```diff
@@ -20,10 +20,10 @@ def tick(self):
             self.dead = True
 
         # update x,y based on vel and facing
-        dx = math.sin(self.angle) * self.velocity
-        dy = math.cos(self.angle) * self.velocity
-        x += dx
-        y += dy
+        dx = math.sin(self.facing) * self.vel
```
Comment (scarletnguyen13) [inline IMP/creatures.py:23]: Let's revert this change later

File: IMP/board.py [hunk 2]
```diff
@@ -2,30 +2,91 @@
...
+        for f in self.food:
+            if random.random() > self.foodspawnthresh:
+                x, y = f.x + self.random.randint() * 10, f.y + self.random.randint() * 10
+                if x > 0 and x < self.width and y > 0 and y < self.height:
+                    self.food.append(food.Food(x, y, val=random.random(0, 3)))
+
+        # !!! check for collisions
+
+    def closest(self, x, y, size):
+        prey_min_r = 1e10
```
Comment (scarletnguyen13) [inline IMP/board.py:28]: use math.inf

File: IMP/board.py [hunk 3]
```diff
@@ -2,30 +2,91 @@
...
+        nets = []
+        g_l = []
+        for genome_id, genome in genomes:
+            genome.fitness = 0  # start with fitness level of 0
+            net = neat.nn.FeedForwardNetwork.create(genome, config)
+            nets.append(net)
+            self.creatures.append(creatures.Creature(x=random.randint(0, self.width), y=random.randint(0, self.height)))
+            g_l.append(genome)
+
+        while self.creatures and self.ticks_total < 1800:
```
Comment (scarletnguyen13) [inline IMP/board.py:66]: Can you store 1800 in a variable for clarification please? :)

Comment (jackyzha0) [reply IMP/board.py:66]: got it

File: IMP/train.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    p.add_reporter(stats)
+
+    # Run for up to 50 generations.
+    b = board.Board(500, 500)
+    top = p.run(b.sim_one_gen, 50)
+
+    # show final stats
+    print('\nBest genome:\n{!s}'.format(top))
+
+def render(self, screen):
```
Comment (scarletnguyen13) [inline IMP/train.py:24]: Duplicated function, we don't need this here

**Review by scarletnguyen13 — APPROVED** [review]
Cool beans 💯

**Pull request merged**